src/models/concrete_syntax.rs

```diff
@@ -0,0 +1,89 @@
+/*
+ Copyright (c) 2023 Uber Technologies, Inc.
+
+ <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ except in compliance with the License. You may obtain a copy of the License at
+ <p>http://www.apache.org/licenses/LICENSE-2.0
+
+ <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ express or implied. See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+use crate::models::matches::Range;
+use derive_builder::Builder;
+use regex::Regex;
+use std::collections::{HashMap, VecDeque};
+use tree_sitter::{Node, Parser, TreeCursor};
+
+use crate::models::capture_group_patterns::MetaSyntax;
+use crate::models::language::PiranhaLanguage;
+use crate::models::matches::Match;
+use crate::utilities::tree_sitter_utilities::get_all_matches_for_query;
+
+pub(crate) fn get_all_matches_for_metasyntax(
+  node: &Node, source_code: String, meta: &MetaSyntax, recursive: bool,
+  _replace_node: Option<String>,
+) -> (Vec<Match>, bool) {
+  let vec_deque = vec![node.clone()].into_iter().collect::<VecDeque<_>>();
+  get_all_matches_for_metasyntax_aux(vec_deque, source_code, meta, _replace_node)
+}
+
+pub(crate) fn get_all_matches_for_metasyntax_aux(
+  mut nodes: VecDeque<Node>, source_code: String, meta: &MetaSyntax, _replace_node: Option<String>,
```

Comment:
lets also think how we can account for `_replace_node`

Since we have no way to name the outermost node, we can have user insert `*`   to replace the entire expression or the appropriate tag name

Reply:
Sounds good!